content/advent-2017/building-medical-reports-in-go.md

```diff
@@ -0,0 +1,241 @@
+
+	r.drawLine(fs, "<b>Método</b>", 10)
+
+	// MultiCell(width, height, content, border, align, fill)
+	r.document.MultiCell(0, 5, r.encodingFunc(content), "", "", false)
+	r.lineStroke()
+}
+```
+
+We just set the font size for that section, build a text for it and add it to the document using the `MultiCell` function. Easy! No x/y alignment, text limit, etc. The library will handle it based on our previous configuration at `NewReport`.
+
+Going back to the main function, at the end we write the final document back to the `ResposeWriter` and we're done!
+
+Below you could see the result of this POC: a report 99% similar to what we have in python in just 351 LOC:
+
+![](/postimages/advent-2017/building-medial-reports-in-go/poc-page-1.png)
+![](/postimages/advent-2017/building-medial-reports-in-go/poc-page-2.png)
+
+## From Python to Go
+
+As we saw previously, we need to change how Python and Go communicate each other. Instead send an `application/pdf` as before, now the python API need to serve all the data needed to construct the report on the Go side.
+
+We created a new endpoint to serve this new information as we can see below:
+
+![](/postimages/advent-2017/building-medial-reports-in-go/api-v2-diagram.png)
+
+With the API ready we changed how our Go application called it. Instead call the API to build the report now we're calling the model endpoint in order to get the metadata necessary to construct it in our end.
+
+```go
+model, err := api.GetModel(code, user)
+
+if err != nil {
+    errorMessage := "unable to get report model"
+    logrus.WithFields(logrus.Fields{
+        "code":  code,
+        "user":  user,
+        "error": err,
+    }).Error(errorMessage)
+    http.Error(w, errorMessage, http.StatusInternalServerError)
+}
+```
+
+After that we pass this data to our `report` object:
+
+```go
+if err := report.New(model).Output(w); err != nil {
+    errorMessage := "unable to generate report"
+    logrus.WithFields(logrus.Fields{
+        "code":  code,
+        "error": err,
+    }).Error(errorMessage)
+    http.Error(w, errorMessage, http.StatusInternalServerError)
+}
+```
+
+Did you see how this code is very similar which our POC? Yes, you got this right, we just basically copy&paste these code from POC to our application and just remove the hardcode part related to how we build each section. The `Output` function is pretty the same:
+
+```go
+func (r *Report) Output(w io.Writer) error {
+	if err := r.Error(); err != nil {
+		r.document.SetError(err)
+		return r.document.Output(w)
+	}
+
+	r.patientHeader()
+	r.diagnostic()
+	r.geneList()
+	r.method()
+	r.qualityFlags()
+	r.vusSection()
+	r.comments()
+	r.additionalInformation()
+
+	return r.document.Output(w)
+}
+```
+
+## Bonus: Unit Test For PDF
+
+Your methods to build each PDF's section is fully tested, but if I change something like font color, an image position or another visual thing? How can I get feedback about it?
```

Comment:
Suggestions: change `section is fully tested` to `section are fully tested`, and `but if I` to `but what if I`.